packages/plugin-heading/Heading.ts

```diff
@@ -17,4 +55,29 @@ export class Heading extends JWPlugin {
             return [context, parsingMap];
         }
     }
+    /**
+     * Change the formatting of the nodes in given range to Heading.
+     *
+     * @param params
+     */
+    applyParagraph(params: ParagraphParams): void {
+        let heading = params.nodeCreator(); // todo: make sure to keep attributes
+        if (heading.is(HeadingNode)) {
+            const range = params.range || this.editor.vDocument.selection.range;
+            let nodesToConvert: VNode[];
+            if (range.isCollapsed()) {
+                nodesToConvert = [range.start.parent];
+            } else {
+                const selectedLeaves = range.selectedNodes(node => !node.children.length);
+                nodesToConvert = distinct(
+                    selectedLeaves.map(leaf => (leaf.atomic ? leaf.parent : leaf)),
+                );
+            }
+            for (const node of nodesToConvert) {
+                node.before(heading);
+                node.mergeWith(heading);
+                heading = params.nodeCreator();
```

Comment:
If the parameter is `level` rather `nodeCreator`, you can just create the new node at the top of the `for`. Actually, you could have done that even now, since you are not using `heading` before the `for`. It would have been easier to read than creating the "next" heading node at the end of the "current" loop iteration.